VoodooInput/VoodooInputSimulator/VoodooInputSimulatorDevice.cpp

```diff
@@ -214,77 +213,63 @@ void VoodooInputSimulatorDevice::constructReportGated(const VoodooInputEvent& mu
     }
 
     if (input_active)
-        input_report.TouchActive = 0x3;
+        input_report->TouchActive = 0x3;
     else
-        input_report.TouchActive = 0x2;
+        input_report->TouchActive = 0x2;
     
-    int total_report_len = (9 * multitouch_event.contact_count) + 12;
-    IOBufferMemoryDescriptor* buffer_report = IOBufferMemoryDescriptor::inTaskWithOptions(kernel_task, 0, total_report_len);
-
     if (!is_error_input_active) {
-      buffer_report->writeBytes(0, &input_report, total_report_len);
-      handleReport(buffer_report, kIOHIDReportTypeInput);
+        handleReport(input_report_buffer,kIOHIDReportTypeInput);
```

Comment:
Again you should maintain track of the data length after writing bytes to `input_report`. Any problem with `input_report_buffer->setLength()`?

Reply:
Since i'm erasing memory with memset() , adjusting length is not necessary.Nulled memories are ignored anyway.

Reply:
Wouldn't it be safer to simply set it, as `handleReport` processes the report explicitly based on the data length?